CMakeLists.txt

```diff
@@ -407,6 +408,18 @@ else()
     )
 endif()
 
+if((UMF_BUILD_GPU_TESTS OR UMF_BUILD_GPU_EXAMPLES) AND UMF_BUILD_CUDA_PROVIDER)
+    find_package(CUDA REQUIRED cuda)
+    if(CUDA_FOUND)
```

Comment:
I was wrong. The `*_FOUND` variable is set by `pkg_check_modules`. Our `FindLIB*` scripts set the `*_LIBRARIES` variable, so you should use:
```
if(CUDA_LIBRARIES)
```
instead of `if(CUDA_FOUND)` here.